Tools/AP_Periph/esc_apd_telem.h

```diff
@@ -0,0 +1,61 @@
+/*
+  ESC Telemetry for APD ESC.
+ */
+
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+
+#ifdef HAL_PERIPH_ENABLE_ESC_APD
+
+class ESC_APD_Telem {
+public:
+    ESC_APD_Telem();
+
+    void init(AP_HAL::UARTDriver *_uart, float num_poles);
+    bool update();
+
+    struct telem {
+        uint32_t error_count;
+        float voltage;
+        float current;
+        float temperature; // kelvin
+        int32_t rpm;
+        uint8_t power_rating_pct;
+    };
+
+    const telem &get_telem(void) {
+        return decoded;
+    }
+
+private:
+    AP_HAL::UARTDriver *uart;
+
+    union {
+        struct PACKED {
+            uint16_t voltage;
+            uint16_t temperature;
+            int16_t bus_current;
+            uint16_t reserved0;
+            uint32_t erpm;
+            uint16_t input_duty;
+            uint16_t motor_duty;
+            uint16_t reserved1;
+            uint16_t checksum; // 16 bit fletcher checksum
+            uint16_t stop; // should always be 65535 on a valid packet
+        } packet;
+        uint8_t bytes[22];
+    } received;
+    static_assert(sizeof(received.packet) == sizeof(received.bytes), "The packet must be the same size as the raw buffer");
+
+    uint8_t len;
+
+    struct telem decoded;
+
+    float pole_count;
```

Comment:
`float`?!  Is this some new sort of physics?

Reply:
Float was more useful given the resultant types. Either way more egregiously it was supposed to be a parameter, it got hard coded for testing...

Reply:
Okay, I've added the parameters, but I'm sorta a fan here of just using slightly more RAM while saving the floating point conversion... If it is horridly offensive I can change it.